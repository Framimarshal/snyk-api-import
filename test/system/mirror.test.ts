import { exec } from 'child_process';
import { sep } from 'path';
import { deleteFiles } from '../delete-files';

const main = './dist/index.js'.replace(/\//g, sep);

describe('`snyk-api-import mirror <...>`', () => {
    const OLD_ENV = process.env;
    process.env.GITHUB_TOKEN = process.env.GH_TOKEN;
    const GROUP_ID = process.env.TEST_GROUP_ID as string;
    const ORG_ID = process.env.TEST_ORG_ID as string;
  
    afterAll(async () => {
      process.env = { ...OLD_ENV };
    });
    it('Shows help text as expected', (done) => {
      exec(`node ${main} mirror help`, (err, stdout) => {
        if (err) {
          throw err;
        }
        expect(err).toBeNull();
        expect(stdout).toMatchSnapshot()
      }).on('exit', (code) => {
        expect(code).toEqual(0);
        done();
      });
    }, 20000);

  });