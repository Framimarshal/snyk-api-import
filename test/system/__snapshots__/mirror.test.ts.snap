// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`snyk-api-import mirror <...>\` Shows help text as expected 1`] = `
"index.js mirror

mirror the 3 commands: orgs:data, orgs:create and import. Get the data from the
source, create matching organisation into snyk and import the data

Options:
  --version              Show version number                           [boolean]
  --help                 Show help                                     [boolean]
  --snykGroupId          Public id of the group in Snyk (available on group
                         settings)                                    [required]
  --source               The source of the targets to be imported e.g. Github,
                         Github Enterprise, Gitlab, Azure. This will be used to
                         make an API call to list all available entities per org
    [required] [choices: \\"github\\", \\"github-enterprise\\", \\"gitlab\\", \\"azure-repos\\",
                      \\"bitbucket-server\\", \\"bitbucket-cloud\\"] [default: \\"github\\"]
  --sourceUrl            Custom base url for the source API that can list
                         organizations (e.g. Github Enterprise url)
  --snykIntegrationName  The configured integration type on the created Snyk Org
                         to use for generating import targets data. Applies to
                         all targets.
               [choices: \\"github\\", \\"github-enterprise\\", \\"gitlab\\", \\"azure-repos\\",
                                          \\"bitbucket-server\\", \\"bitbucket-cloud\\"]
  --sourceOrgPublicId    Public id of the organization in Snyk that can be used
                         as a template to copy all supported organization
                         settings.
"
`;
